promotion/datastore.go

```diff
@@ -691,3 +703,80 @@ limit 1`
 
 	return attempted, nil
 }
+
+// This code can be deleted once https://github.com/brave-intl/bat-go/issues/263 is addressed.
+
+// GetOrder queries the database and returns an order
+func (pg *Postgres) GetOrder(orderID uuid.UUID) (*Order, error) {
+	statement := "SELECT * FROM orders WHERE id = $1"
+	order := Order{}
+	err := pg.DB.Get(&order, statement, orderID)
+	if err == sql.ErrNoRows {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	foundOrderItems := []OrderItem{}
+	statement = "SELECT * FROM order_items WHERE order_id = $1"
+	err = pg.DB.Select(&foundOrderItems, statement, orderID)
+
+	order.Items = foundOrderItems
+	if err != nil {
+		return nil, err
+	}
+
+	return &order, nil
+}
+
+// UpdateOrder updates the orders status.
+// 	Status should either be one of pending, paid, fulfilled, or canceled.
+func (pg *Postgres) UpdateOrder(orderID uuid.UUID, status string) error {
+	_, err := pg.DB.Exec(`UPDATE orders set status = $1, updated_at = CURRENT_TIMESTAMP where id = $2`, status, orderID)
+
+	if err != nil {
```

Comment:
Maybe validate also that at least one record was updated?

Reply:
Updated! https://github.com/brave-intl/bat-go/pull/264/commits/afba0039ca66688d6f6675850e093e1636b5a2a4